pkg/diagnostics/diagnostic_bundle.go

```diff
@@ -69,7 +69,7 @@ func newDiagnosticBundleManagementCluster(af AnalyzerFactory, cf CollectorFactor
 		writer:           writer,
 	}
 
-	b.WithDefaultCollectors().WithDefaultAnalyzers().WithManagementCluster(true)
+	b.WithDefaultCollectors().WithDefaultAnalyzers().WithManagementCluster(true).WithDatacenterConfig(spec.Cluster.Spec.DatacenterRef)
```

Comment:
the one issue I foresee here (an edge case) is that if we collect and analyze from a bootstrap cluster which the CAPI components have been moved off of, the `WithDataCenterConfig` analyzers will still expect them to be present and alert on it. don't have a solution for that at the moment as the diagnostic bundle has no awareness of the 'phase' a cluster is in, but wanted to call it out for further thought

Reply:
Agreed. It seems like to have a comprehensive set of artifacts for troubleshooting, we need to either be surgical and phase-aware when we're collecting them, or just do a full sweet of all the possible artifacts and expect some to be missing. I feel like the latter is more likely to get us the artifacts we need without coupling this code with the main eks-a logic

Reply:
is this method supposed to handle both bootstrap cluster before/after `clusterapi move`, eks-a management cluster in the management and workload setting? If so, why below are not part of the analyzers?
```
		WithGitOpsConfig(spec.GitOpsConfig).
		WithOidcConfig(spec.OIDCConfig).
		WithExternalEtcd(spec.Cluster.Spec.ExternalEtcdConfiguration).
		WithMachineConfigs(provider.MachineConfigs(spec)).
```

Reply:
It should probably handle the cases before/after `move`. I would imagine the reason is "because we've never needed it before"? But I'd prefer to redirect this conversation to a new issue, since I don't think it's related to my specific change to unblock the e2e tests.